lib/internal/cluster/master.js

```diff
@@ -49,7 +49,8 @@ cluster.setupMaster = function(options) {
     silent: false
   };
   settings = util._extend(settings, cluster.settings);
-  settings = util._extend(settings, options || {});
+  if (options)
```

Comment:
One option here would be `Object.assign(settings, cluster.settings, options)`. I know `util._extend()` is faster, but maybe not if you have to call it twice (needs benchmarked)?